### PR TITLE
python312Packages.shiny: init at 0.10.2

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

Fix:
- #325433

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/htmltools/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "posit-dev";
+    repo = "py-htmltools";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-H0M9dY8CNQAMEGEGHhPIWEYRmk4omCuVFgJUg8ef8Zw=";
+  };
+
+  build-system = [ setuptools ];
+  dependencies = [ typing-extensions ];
```
Comment (applePrincess) [inline pkgs/development/python-modules/htmltools/default.nix:24]: `packaging` is missing from dependencies.

**Review by applePrincess — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/shiny/default.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    pytest-playwright
+    pytest-xdist
+    pytest-timeout
+    pytest-rerunfailures
+    pytest-cov
+    pandas
+  ];
+
+  meta = {
+    changelog = "https://github.com/py-shiny/py-shiny/blob/${src.rev}/CHANGELOG.md";
```
Comment (applePrincess) [inline pkgs/development/python-modules/shiny/default.nix:73]:
```suggestion
    changelog = "https://github.com/posit-dev/py-shiny/blob/${src.rev}/CHANGELOG.md";
```

Comment (Sigmanificient) [reply pkgs/development/python-modules/shiny/default.nix]: whoops, copy paste owner fail, i guess

**Comment by applePrincess** [comment]
Since #321316 has been merged, could you remove htmltools related commit?

**Comment by Sigmanificient** [comment]
oh well that pr was mislabeled :sweat_smile:

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 326910` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.shiny</li>
    <li>python311Packages.shiny.dist</li>
    <li>python312Packages.shiny</li>
    <li>python312Packages.shiny.dist</li>
  </ul>
</details>

**Review by applePrincess — APPROVED** [review]

File: pkgs/development/python-modules/shiny/default.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    pytest-playwright
+    pytest-xdist
+    pytest-timeout
+    pytest-rerunfailures
+    pytest-cov
+    pandas
+  ];
+
+  meta = {
+    changelog = "https://github.com/posit-dev/py-shiny/py-shiny/blob/${src.rev}/CHANGELOG.md";
```
Comment (natsukium) [inline pkgs/development/python-modules/shiny/default.nix:73]:
```suggestion
    changelog = "https://github.com/posit-dev/py-shiny/blob/${src.rev}/CHANGELOG.md";
```

File: pkgs/development/python-modules/shiny/default.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  pythonImportsCheck = [ "shiny" ];
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-asyncio
+    pytest-playwright
+    pytest-xdist
+    pytest-timeout
+    pytest-rerunfailures
+    pytest-cov
```
Comment (natsukium) [inline pkgs/development/python-modules/shiny/default.nix:68]:
```suggestion
```
Please remove the coverage-related package.

File: pkgs/development/python-modules/shiny/default.nix [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  watchfiles,
+  websockets,
+
+  pytestCheckHook,
+  pytest-asyncio,
+  pytest-playwright,
+  pytest-xdist,
+  pytest-timeout,
+  pytest-rerunfailures,
+  pytest-cov,
```
Comment (natsukium) [inline pkgs/development/python-modules/shiny/default.nix:27]:
```suggestion
```

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**